pkg/kvstore/etcd.go

```diff
@@ -580,6 +590,29 @@ func connectEtcdClient(ctx context.Context, config *client.Config, cfgPath strin
 		}),
 	}
 
+	initialLimit := clientOptions.RateLimit
+	// If BootstrapRateLimit and BootstrapComplete are provided, set the
+	// initial rate limit to BootstrapRateLimit and apply the standard rate limit
+	// once the caller has signaled that bootstrap is complete by closing the channel.
+	if clientOptions.BootstrapRateLimit > 0 && opts != nil && opts.BootstrapComplete != nil {
+		ec.logger.WithField(logfields.EtcdQPSLimit, clientOptions.BootstrapRateLimit).Info("Setting client QPS limit for bootstrap")
+		initialLimit = clientOptions.BootstrapRateLimit
+		go func() {
+			select {
+			case <-ctx.Done():
```

Comment:
I'd replace this with `ec.client.Ctx().Done()`, as ctx is not guaranteed to be closed if the `client.Close()` gets called.